src/mscorlib/shared/System/IO/PathInternal.Unix.cs

```diff
@@ -55,23 +57,29 @@ internal static string NormalizeDirectorySeparators(string path)
                 }
             }
 
-            if (normalized) return path;
+            if (normalized)
+                return path;
 
-            StringBuilder builder = new StringBuilder(path.Length);
-
-            for (int i = 0; i < path.Length; i++)
+            fixed (char* f = path)
```

Comment:
You do not need to use unsafe code and pin it anymore now. You can just pass the string through as string.